### DISCUSSION TITLE
Unable to Reset Form Field with controller

### DISCUSSION THREAD

**Discussion by mernaff18** [react-hook-form/react-hook-form#8016] [Q&A] [answered]
I am using external library with React hook form Controller. When i am trying to reset the form its not happening..

https://codesandbox.io/s/dreamy-forest-1y6m23

**Comment by Moshyfawn** [comment, marked as answer]
Your `ReactFlagsSelect` is using the external React state you've setup via `useState`, which RHF has no control over.
Keep your controlled input's values inside the RHF state instead, just like I've showed you in #8007
```js
render={({ field: { onChange, onBlur, value, ref } }) => (
  <ReactFlagsSelect onSelect={onChange} selected={value} />
)}
```

Also, it's best to use [the RHF `reset` method](https://react-hook-form.com/api/useform/reset) instead of the native form event's one to properly reset your controlled inputs.
```diff
+ const { reset } = useForm()

const onSubmit = (data, e) => {
  console.log(data);
- e.target.reset();
+ reset()
};
```

See [this working example](https://codesandbox.io/s/immutable-waterfall-smjsj7?file=/src/App.js)
